tests/math.py

```diff
@@ -56,8 +56,8 @@ def eval_math(car, cdr):
     return None
 
 # Rewrite rules, not all are currently used since gaurds aren't in snake-egg yet
-a, b, c, x, f, g, y = vars("a b c x f g y")
-list_rules = [
+a, b, c, x, f, g, y = vars("a b c x f g y") # type: ignore
+list_rules: List[List[Any]] = [
```

Comment:
We need this type signature, otherwise mypy will complain later down when creating the rewrites because it doesn't know the inner types.

If we wanted to avoid this manual imprecise type here, we could instead just create the rewrite list directly instead of first making lists of items then creating it.